services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowConnectedDeviceSpec.groovy

```diff
@@ -93,43 +129,95 @@ srcLldpDevices=#newSrcEnabled, dstLldpDevices=#newDstEnabled"() {
         validateLldpRulesOnSwitch(updatedFlow, true)
         validateLldpRulesOnSwitch(updatedFlow, false)
 
+        when: "Two devices send lldp packet on each flow endpoint"
+        def srcData = LldpData.buildRandom()
+        def dstData = LldpData.buildRandom()
+        def tgService = traffExamProvider.get()
+        withPool {
+            [[flow.source, srcData], [flow.destination, dstData]].eachParallel { endpoint, lldpData ->
+                new ConnectedDevice(tgService, topology.getTraffGen(endpoint.datapath), endpoint.vlanId).withCloseable {
+                    it.sendLldp(lldpData)
+                }
+            }
+        }
+
+        then: "Getting connecting devices shows corresponding devices on each endpoint according to updated status"
+        Wrappers.retry(3, 0.5) { //need some time for devices to appear
+            with(northbound.getFlowConnectedDevices(flow.id)) {
+                it.source.lldp.size() == (newSrcEnabled ? 1 : 0)
```

Comment:
`with` supports implicit assertions as well. I don't think that `verifyAll` soft verifications are required here, since all verification pretty much depend on each other and order they happen